### PR TITLE
cdb: Change all foreign key that refers from tx_hash to id.

### PR DESCRIPTION
**Pull request by Benzbeeb**
## Implementation details
1. Change all foreign key that refers from tx_hash to id.
2. Fix flusher handler
---

Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [x] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [x] The pull request includes a description of the implementation/work done in detail.
- [ ] The pull request includes any and all appropriate unit/integration tests
- [x] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [x] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

### CODE REVIEW

File: flusher/flusher/handler.py [hunk 1]
```diff
@@ -47,20 +48,38 @@ def handle_set_account(self, msg):
         )
 
     def handle_set_data_source(self, msg):
+        if "tx_hash" in msg:
+            msg["transaction_id"] = self.conn.execute(
+                select([transactions.c.id]).where(transactions.c.hash == msg["tx_hash"])
+            ).scalar()
+        else:
+            msg["transaction_id"] = None
+        del msg["tx_hash"]
```
Comment (sorawit) [inline flusher/flusher/handler.py:71]: I don't think you can delete a key if it does not exist.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: flusher/flusher/handler.py [hunk 2]
```diff
@@ -47,20 +48,38 @@ def handle_set_account(self, msg):
         )
 
     def handle_set_data_source(self, msg):
+        if msg["tx_hash"] is not None:
+            msg["transaction_id"] = self.conn.execute(
```
Comment (taobun) [inline flusher/flusher/handler.py]: Can we create a helper function to get transaction_id instead of copy these lines of code to be the same work?

**Review by sorawit — APPROVED** [review]
Thanks!

File: flusher/flusher/db.py [hunk 3]
```diff
@@ -155,8 +155,8 @@ def Column(*args, **kwargs):
     "reports",
     metadata,
     Column("request_id", sa.Integer, sa.ForeignKey("requests.id"), primary_key=True),
+    Column("transaction_id", sa.Integer, sa.ForeignKey("transactions.id")),
     Column("validator_id", sa.Integer, sa.ForeignKey("validators.id"), primary_key=True),
-    Column("tx_hash", CustomBase64, sa.ForeignKey("transactions.hash")),
     Column("reporter", sa.String),
```
Comment (taobun) [inline flusher/flusher/db.py]: Change to reporter_id

**Review by taobun — CHANGES_REQUESTED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**